### ISSUE TITLE
Screenshot cannot be written on disk if web page title contains forbidden characters

### ISSUE THREAD

**Issue by syldub69** [M-Reimer/savescreenshot#18]
Hi,

I'm using Firefox 76 on Windows 10 64-bit.

I am using this standard filename format for Save Screenshot extension:
%t - %Y-%m-%d %Hh%Mm%S

If the web page title (%t) does contain some characters which are not allowed on Windows for filename, the screenshot file cannot be written on disk (there is no error message though... nothing happen).

The list of forbidden characters for filename on Windows are :
\ / : * ? " < > |

Could you implement a feature which will allow to automatically remove these characters ? (or  maybe best it to replace them by a configurable character in the settings)

**Comment by M-Reimer** [comment]
I was pretty sure that I already filter them.
Do you have a demo URL?

**Comment by syldub69** [comment]
Yes, you can try these for instance :
https://www.allobebe.fr/raclette-tartiflette-fondue-grossesse.html
https://fr.aliexpress.com

**Comment by M-Reimer** [comment]
The problem are not "forbidden characters" but groups of space characters that are a problem for some reason.

But I'm sure this is not the last problem with filenames...

The way they are handled by Mozilla internally just sucks.

**Comment by syldub69** [comment]
What you exactly mean by "groups of space characters"?

**Comment by M-Reimer** [comment]
For example two space characters in a row.

**Comment by syldub69** [comment]
The title of the 2 web pages that I did communicate before as examples don't have more than one consecutive white space, that is why I don't understand your remark. 
We have :
Raclette, tartiflette, fondue autorisés pendant la grossesse ? - www.allobebe.fr
FR.AliExpress | aliexpress en français - Acheter en ligne des produits de qualité à petits prix en Chine. - AliExpress - AliExpress - fr.aliexpress.com

To my mind, the issue seems to originate from the fact that first one contains "?" and second one contains "|".

**Comment by M-Reimer** [comment]
I did debug this and with my fix it works now, so you can trust me on this :stuck_out_tongue: 

"?" and "|" were replaced almost from the start of this Add-on.

But if you remove the "|" in the "AliExpress title, then the space character in front of it and after it end up as two space characters in a row.

For whatever reason Mozilla doesn't like that in filenames.

**Comment by syldub69** [comment]
ok, now I understand well what you explained! 😃
I didn't know that the "forbidden" characters were removed by the extension. That is why at the beginning I suggested to replace "forbidden" character instead of removing them.
Actually the "two space characters in a row issue" is not a Firefox dependent issue, it is a HTML feature which work like it is defined. In HTML, whitespace is largely ignored — whitespace in between words is treated as a single character, and whitespace at the start and end of elements and outside elements is ignored.
See https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace
